utils/demangle.c

```diff
@@ -1464,6 +1465,15 @@ static int dd_source_name(struct demangle_data *dd)
 	if (add_name)
 		dd_append_len(dd, p, num);
 	__dd_consume_n(dd, num, NULL);
+	while (true) {
+		separator = strstr(dd->new, "..");
+		if (separator) {
+			separator[0] = ':';
+			separator[1] = ':';
+		}
+		else
+			break;
+	}
```

Comment:
Another concern is that it's suboptimal.  It checks ".." whenever it calls `dd_source_name()`.  I think we can do it only when the dollar sign was found.  Also it doesn't need to search the whole (new) string.  Could we check the newly added region only?

Reply:
I'm not sure that the ".." is always inside the <>. I'll try to find a counterexample, and if there's nothing, I'll fix this commit.

Reply:
I collect the LLVM IR of large-scale projects and search the pattern like this:
```
$ cat *.ll | grep "define[^w]*\.\." | grep -v "$"
```
and it returns nothing.
So ".." only generated with a dollar sign is quite a safe guess. I'll work on the optimal solution.